Installation/include/CGAL/version_checker.h

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2023 GeometryFactory.
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org)
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: LGPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : -
+
+#ifndef CGAL_VERSION_CHECKER_H
+#define CGAL_VERSION_CHECKER_H
+
+#include <CGAL/version_macros.h>
+
+// All files including this header are meant to work with a given version of CGAL
+// When using forked headers, set the following macro to the version of CGAL
+// you want to use.
+
+//// Set the 3 following macros to the version of CGAL you want to use
+//#define CGAL_AUTHORIZED_VERSION_MAJOR 6
+//#define CGAL_AUTHORIZED_VERSION_MINOR 0
+//#define CGAL_AUTHORIZED_VERSION_PATCH 0
+
+// Set the following macros to 1 to get a warning/an error
+// when using a bad version of CGAL
+#define CGAL_VERSION_CHECKER_ERROR 0
+#define CGAL_VERSION_CHECKER_WARNING 0
+
+#define CGAL_AUTHORIZED_VERSION_STR CGAL_STR(CGAL_AUTHORIZED_VERSION_MAJOR) "." \
+                                    CGAL_STR(CGAL_AUTHORIZED_VERSION_MINOR) "." \
+                                    CGAL_STR(CGAL_AUTHORIZED_VERSION_PATCH)
+
+
+// Check that the version of CGAL used is the one expected
+#if  CGAL_AUTHORIZED_VERSION_MAJOR != CGAL_VERSION_MAJOR \
```

Comment:
AUTHORIZED -> EXPECTED?

Reply:
I'd put something like "compatible" and replace "bad version" / "wrong version" / "authorized version" to "compatible" / "incompatible"